pilot/pkg/xds/delta.go

```diff
@@ -287,9 +288,7 @@ func (s *DiscoveryServer) processDeltaRequest(req *discovery.DeltaDiscoveryReque
 
 	subs := sets.New(req.ResourceNamesSubscribe...).Delete("*")
 	// InitialResourceVersions are essential subscriptions on the first request, since we don't care about the version
-	for k := range req.InitialResourceVersions {
-		subs.Insert(k)
-	}
+	subs.InsertAll(maps.Keys(req.InitialResourceVersions)...)
```

Comment:
This is a performance regression, it allocates an intermediate slice we do not need. IMO not worth a 1 line cleanup